### PR TITLE
Add support for multiple models for one config in auto classes

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR adds support for having multiple models with the same config be in the same auto class. For instance `FunnelBaseModel` and `FunnelModel` are both valid models for the class `AutoModel` but since they both rely on `FunnelConfig`, only `FunnelModel` was in the model mapping for `AutoModel`.

The mechanism when loading changes slightly: if the mapping finds a tuple for the config at hand, it will look into the `architectures` field and return the model in the tuple corresponding to the architecture found there, or the first model of the tuple as a default.

While diving into this, I realized that TF and Flax pretrained models do not populate the field architectures of their configs, so I added support for this.

The rest of the changes are need to adapt to the fact some model mappings now can have tuple values.

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
This looks great! Thanks for handling it.

File: src/transformers/models/auto/configuration_auto.py [hunk 1]
```diff
@@ -247,29 +247,38 @@
 )
 
 
+def _get_class_name(model_class):
+    if isinstance(model_class, (list, tuple)):
+        return " or ".join([f":class:`~transformers.{c.__name__}`" for c in model_class])
+    return f":class:`~transformers.{model_class.__name__}`"
```
Comment (sgugger) [inline src/transformers/models/auto/configuration_auto.py:253]: This change and below are for the docstrings to properly render a model type with several models possible: 
image

Comment (LysandreJik) [reply src/transformers/models/auto/configuration_auto.py:253]: oh this is very nice

File: tests/test_modeling_flax_bert.py [hunk 2]
```diff
@@ -29,6 +29,7 @@
         FlaxBertForNextSentencePrediction,
         FlaxBertForPreTraining,
         FlaxBertForQuestionAnswering,
+        FlaxBertForSequenceClassification,
```
Comment (LysandreJik) [inline tests/test_modeling_flax_bert.py:32]: Nice catch!

**Review by LysandreJik — APPROVED** [review]
Great, this looks good to me! Thanks for handling it @sgugger

File: tests/test_modeling_flax_bert.py (hunk 2 above)
Comment (sgugger) [reply tests/test_modeling_flax_bert.py:32]: Not mine, it's the script. I'll let it know you liked ;-)

**Pull request merged**